### PR TITLE
[SPARK-19670] [SQL] [TEST] Enable Bucketed Table Reading and Writing Testing Without Hive Support

### PR DESCRIPTION
**Pull request by gatorsmile**
### What changes were proposed in this pull request?
Bucketed table reading and writing does not need Hive support. We can move the test cases from `sql/hive` to `sql/core`. After this PR, we can improve the test case coverage. Bucket table reading and writing can be tested with and without Hive support.

### How was this patch tested?
N/A

### CODE REVIEW

**Comment by gatorsmile** [comment]
cc @cloud-fan @tejasapatil

File: sql/core/src/test/scala/org/apache/spark/sql/sources/BucketedWriteSuite.scala [hunk 1]
```diff
@@ -20,19 +20,29 @@ package org.apache.spark.sql.sources
...
+import org.apache.spark.sql.test.{SharedSQLContext, SQLTestUtils}
 
-class BucketedWriteSuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
+class BucketedWriteWithoutHiveSupportSuite extends BucketedWriteSuite with SharedSQLContext {
+  protected override def beforeAll(): Unit = {
+    super.beforeAll()
+    assume(spark.sparkContext.conf.get(CATALOG_IMPLEMENTATION) == "in-memory")
+  }
+
+  override protected def fileFormatsToTest: Seq[String] = Seq("parquet", "json")
```
Comment (tejasapatil) [inline sql/core/src/test/scala/org/apache/spark/sql/sources/BucketedWriteSuite.scala]: curious : why is `orc` not in this list ?

**Comment by tejasapatil** [comment]
LGTM with a minor nit

File: sql/core/src/test/scala/org/apache/spark/sql/sources/BucketedWriteSuite.scala (hunk 1 above)
Comment (gatorsmile) [inline sql/core/src/test/scala/org/apache/spark/sql/sources/BucketedWriteSuite.scala]: `orc` is not available in sql/core package. : (

**Comment by gatorsmile** [comment]
retest this please

**Comment by gatorsmile** [comment]
Thanks! Merging to master.

**Pull request closed without merging**